zonemaster/zonemaster-gui
# add runtime configuration support

## Purpose

Add a runtime configuration file editable after compilation. This is useful to change the app settings (such as the api endpoint) without recompiling the whole thing.

## Context

* Suggested in #219
* Can fix #170 (current configuration make it easy to change the default language / restrict only to certain languages)
* Can make #252 / #250 easier

## Changes

* Add a configuration file `assets/app.config.json` that can be retrieve by the client. This configuration overrides the values of `environment.ts`, so we can treat the `environment.ts` values as default.
* Make the languages a configuration item instead of being hardcoded.

## How to test this PR

* As-is the GUI should behave as usual.
* Change the content of `assets/app.config.json` to be
  ```json
  {
    "defaultLanguage": "fr",
    "enabledLanguages": [ "en", "fr" ],
    "apiEndpoint": "https://zonemaster.fr/api"
  }
  ```
* Restart the GUI, verify that the changes reflect on the GUI.
* Other configuration items can tested, such as the logo or contact address.
